core/src/main/java/com/segment/analytics/kotlin/core/Analytics.kt

```diff
@@ -503,6 +493,24 @@ class Analytics internal constructor(
             decodeFromJsonElement(deserializationStrategy, it)
         }
     }
+
+    /**
+     * Retrieve the settings  in a blocking way.
+     * Note: this method invokes `runBlocking` internal, it's not recommended to be used
+     * in coroutines.
+     */
+    @BlockingApi
+    fun settings(): Settings? = runBlocking {
```

Comment:
need to add a `settings` function to `JavaAnalytics` for compatibility. also unit tests for them

Reply:
ah good catch. will add those